### DISCUSSION TITLE
Combining MPING and JDBC_PING

### DISCUSSION THREAD

**Discussion by ascertrobw** [belaban/JGroups#574] [Q&A] [answered]
Complete newbie here - so apols if asked and answered, or a dumb Q!

Would it ever make sense to config say MPING and JDBC_PING, and what would be the outcomes if so?

I can see this might allow a common config file where some installations could use multicast if supported, and others that couldn't would have the option to fall back to database discovery.

But does anyone know what would happen if an installation had both that were usable, but not for all group members e.g. some members were on a subnet that were multicast discoverable but also had database connectivity, and others were on distributed nodes that only had database discovery. Would JGroups just figure it out and work, or would having both declared cause conflicts/collisions of discovery on the subnet with multicast enabled?

**Comment by belaban** [comment, marked as answer]
On 23.11.21 08:52, RobWalker wrote:
> Complete newbie here - so apols if asked and answered, or a dumb Q!
> 
> Would it ever make sense to config say MPING and JDBC_PING, and what 
> would be the outcomes if so?


Yes, absolutely. In earlier versions, MULTI_PING had to be used, but 
since a certain version of JGroups, this isn't necessary any longer. 
Just add as many discovery protocols as you want to your config!

All discovery protocols will be queried: if async_discovery is true, 
then this will be done asynchronously (1 thread per discovery protocol).


> I can see this might allow a common config file where some installations 
> could use multicast if supported, and others that couldn't would have 
> the option to fall back to database discovery.

That's the idea

> But does anyone know what would happen if an installation had both that 
> were usable, but not for all group members e.g. some members were on a 
> subnet that were multicast discoverable but also had database 
> connectivity, and others were on distributed nodes that only had 
> database discovery. Woudl JGroups just figure it out and work, or would 
> having both declared cause conflicts/collisions of discovery on the 
> subnet with multicast enabled?

This should not be a problem, as the discovery *responses* would include 
the same information (uuids). All discovery protocols share a common 
(conceptual) completable future, which returns as soon as valid results 
have been added.


> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/belaban/JGroups/discussions/574>, or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AADPZXIRH2APOVAPJCG2ZDDUNNB2NANCNFSM5IS2YH3A>.
> Triage notifications on the go with GitHub Mobile for iOS 
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> 
> or Android 
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
> 
> 

-- 
Bela Ban | http://www.jgroups.org

**Reply by ascertrobw** [reply]
That's great. Thanks for the response
